python/feathub/feature_views/sliding_feature_view.py

```diff
@@ -270,6 +307,38 @@ def build(
         )
         return feature_view
 
+    @property
+    def step_size(self) -> timedelta:
+        """
+        Returns the step size of the sliding window features in the feature view.
+        """
+        step_size_set = set(
+            [
+                feature.transform.step_size
+                for feature in self.features
+                if isinstance(feature, Feature)
+                and isinstance(feature.transform, SlidingWindowTransform)
+            ]
+        )
+
+        return step_size_set.pop()
```

Comment:
The code is not very self-explanatory as it assumes that the set size == 1, and this assumption is verified in the constructor.

Can we either cache the step_size obtained in the constructor, or verify set size == 1 in this method?

Same for group_by_keys.